admin/functions-notice.php

```diff
@@ -85,4 +85,75 @@ function aioseop_notice_disable_woocommerce_detected_on_nonpro() {
 		global $aioseop_notices;
 		$aioseop_notices->deactivate_notice( 'woocommerce_detected' );
 	}
+
+	/**
+	 * Set Notice on Activation to Review Plugin
+	 *
+	 * A delayed notice that is set during activation, or initialization (old installs),
+	 * to later display a review/rate AIOSEOP plugin. Delay time: 12 days.
+	 * Delay "...give me a week." 5 days
+	 *
+	 * @since 2.4.2
+	 *
+	 * @global AIOSEOP_Notices $aioseop_notices
+	 *
+	 * @param boolean $update Updates the notice with new content and configurations.
+	 * @param boolean $reset  Notice are re-initiated.
+	 */
+	function aioseop_notice_set_activation_review_plugin( $update = false, $reset = false ) {
+		global $aioseop_notices;
+
+		// TODO Optimize - Create a callback function/method to store most of the configurations (Avoid Database concept).
+		// Dynamic variable could be stored in the database. Config functions could go into a config file/folder.
+		$notice = aioseop_notice_review_plugin();
+
+		if ( $aioseop_notices->insert_notice( $notice ) ) {
```

Comment:
can't this be simplified to 

```
$aioseop_notices->insert_notice( $notice );
if ( $update ) {
...
}
```
as the first `if` is not really doing anything

Reply:
`$aioseop_notices->insert_notice( $notice )` returns a boolean, if it returns false, then it updates.

Insert and Update are 2 different things.

Reply:
yeah, I see. But if you are doing nothing when insert_notice returns true, maybe just simplify it to the below? 
`
if ( ! $aioseop_notices->insert_notice( $notice ) && $update ) {
 // do the update stuff
`

I just feel that commented out code is adding nothing of value but if you intend to put something in there, then your current code is fine.

Reply:
Yea, I do intend to add a footer notice. I forgot this was the notice that was designed differently with an empty IF to use later. I left it there to preserve the alternate code design.